src/compiler/transformers/declarations/emitBinder.ts

```diff
@@ -0,0 +1,914 @@
+                return symbol;
+            }
+        }
+
+        let currentScope = enclosingDeclaration;
+        while (currentScope) {
+            const links = tryGetNodeLinks(currentScope);
+            let symbol = getSymbolFromScope(links?.locals);
+            if (!symbol && (isModuleDeclaration(currentScope) || isSourceFile(currentScope))) {
+                symbol = getSymbolFromScope(links?.symbol?.exports);
+            }
+            if (symbol) return symbol;
+            currentScope = currentScope.parent;
+        }
+        return undefined;
+    }
+
+    function tryGetSymbolFlagsForNode(node: Node): SymbolRegistrationFlags | undefined {
+        return getSymbolFlagsForNode(node as Node & { kind: keyof typeof syntaxKindToSymbolMap; });
+    }
+    function getSymbolFlagsForNode(node: Node & { kind: keyof typeof syntaxKindToSymbolMap; }): SymbolRegistrationFlags {
+        if (node.kind === SyntaxKind.VariableDeclaration) {
+            return getCombinedNodeFlags(node) & NodeFlags.BlockScoped ?
+                syntaxKindToSymbolMap[node.kind].blockScope :
+                syntaxKindToSymbolMap[node.kind].fnScope;
+        }
+        if (node.kind === SyntaxKind.EnumDeclaration) {
+            return isEnumConst(node as EnumDeclaration) ?
+                syntaxKindToSymbolMap[node.kind].const :
+                syntaxKindToSymbolMap[node.kind].regular;
+        }
+        if (node.kind === SyntaxKind.ModuleDeclaration) {
+            return getModuleInstanceState(node as ModuleDeclaration) === ModuleInstanceState.Instantiated ?
+                syntaxKindToSymbolMap[node.kind].value :
+                syntaxKindToSymbolMap[node.kind].namespace;
+        }
+        return syntaxKindToSymbolMap[node.kind];
+    }
+    function bind() {
+        /* eslint-disable no-var */
+        var currentScope: Node = undefined!;
+        var currentFunctionLocalSymbolTable: EmitDeclarationSymbolTable = undefined!;
+        var currentSymbol: EmitDeclarationSymbol = undefined!;
+        var currentLocalSymbolTable: EmitDeclarationSymbolTable = undefined!;
+        var currentExportsSymbolTable: EmitDeclarationSymbolTable | undefined;
+        var postBindingAction: (() => void)[] = [];
+
+        var fileLinks = getNodeLinks(file).symbol = createEmitSymbol();
+        fileLinks.declarations.push(file);
+        setValueDeclaration(fileLinks as Symbol, file);
+        /* eslint-enable no-var */
+        fileLinks.exports = new Map();
+        withScope(file, fileLinks.exports, () => bindEachFunctionsFirst(file.statements));
+        postBindingAction.forEach(fn => fn());
+
+        function createAnonymousEmitSymbol(node: Declaration): EmitDeclarationSymbol {
+            const symbol = createEmitSymbol();
+            symbol.declarations.push(node);
+            setValueDeclaration(symbol as Symbol, node);
+            node.symbol = symbol as Symbol;

```

Comment:
why does this assign to node instead of the node's links?

Reply:
`node.symbol` is accessed in a lot of palces in declaration emit. We fill `declarations` and `valueDeclaration` to allow everything to work.